hardhat.config.js

```diff
@@ -86,27 +86,54 @@ for (const f of fs.readdirSync(path.join(__dirname, 'hardhat'))) {
 const withOptimizations =
   argv.enableGasReport || argv.compileMode === 'production';
 
+
+if (Config.useFaucet){
+  console.log(`Use faucet: ${Config.faucetUrl}`)
+} else {
+  console.log(`Skipping faucet requests: USE_FAUCET set to false`);
+}
+
 const requestFaucet = async (wallet, amount) => {
   if (!Config.useFaucet) {
-    console.log(`Skipping faucet request: USE_FAUCET set to false`);
     return;
   }
-  console.log('Requesting faucet...');
   const data = { amount: amount, wallet: wallet };
-  console.log(`URL: ${Config.faucetUrl}`);
   if (Config.faucetUrl.length === 0) {
     console.log('Unable to request faucet: FAUCET_URL is empty');
     return;
   }
-  console.log(`Wallet = ${data.wallet}, amount = ${data.amount}`);
-  try {
-    const result = await axios.post(Config.faucetUrl, data);
-  } catch (err) {
-    console.log(`Failed to send request to faucet: ${err}`);
+  console.log(`Request '${data.amount}' tokens for wallet ${data.wallet} from faucet`);
+
+  let success = false;
+  for (let i=0; i < 10; i++){
```

Comment:
It'll stop working immediately in environments like DevNet, where the DDoS protection is set. Those environments require one minute's timeout between faucet calls

Reply:
Which ddos protection? If first call success - no any more additional calls. If we get network problem (we sometimes get it) it reconnect us.